numpyro/handlers.py

```diff
@@ -381,7 +383,13 @@ def process_message(self, msg):
                     msg['value'], msg['intermediates'] = msg['fn'].transform_with_intermediates(
                         self.base_param_map[msg['name']])
                 else:
-                    msg['value'] = self.base_param_map[msg['name']]
+                    base_value = self.base_param_map[msg['name']]
```

Comment:
This is pretty neat. I'm not sure if this decomposition of `ComposeTransform` into an invariant first transform and a model dependent second transform will always be valid in the future, but works great for all constraints that we have for now!